### PR TITLE
fix(mcp): reject negative mock_balance in FinStripe get_account_balance

### LINKED ISSUE #329: Bug_120_MUST_FIX: Test Case MCP-BAL-005: Negative mock_balance accepted from config without validation — agents make incorrect payment decisions
**Component:** finbot/mcp/servers/finstripe/server.py → get_account_balance
**Root cause:**
``` python
mock_balance = config.get("mock_balance", DEFAULT_CONFIG["mock_balance"])
# No check: if mock_balance < 0: raise
return {"available_balance": mock_balance, ...}
```

**Steps to reproduce:**
1. Create server with server_config={'mock_balance': -5000}
2. Call get_account_balance

**Expected Behavior:** error — negative balance is invalid 
**Actual Behavior:** available_balance: -5000 returned

**How to execute:**
``` bash
pytest tests/unit/mcp/test_finstripe.py::TestGetAccountBalanceEdgeCases::test_mcp_bal_005_negative_mock_balance_accepted_without_validation -v
```
**Impact:** Tool poisoning via config_json sets negative balance to confuse agent payment decisions.

**Acceptance criteria:**
*TestGetAccountBalanceEdgeCases::test_mcp_bal_005_negative_mock_balance_accepted_without_validation pass

### PR DESCRIPTION
**Pull request by Deez-Automations**
## Summary

Fixes #329.

`get_account_balance` read `mock_balance` straight from `server_config` with no validation at all — a negative value was returned as-is as the account's `available_balance`. Agents (e.g. the Payments Agent) reason over this value when deciding whether a payment is affordable, so a poisoned config with a negative balance could confuse those decisions.

## Fix

Rejects negative balances with a clear error. Zero and ordinary positive balances are unaffected.

## Test plan

- [x] New test file `tests/unit/mcp/test_finstripe.py` — reproduces the exact issue repro steps (`server_config={'mock_balance': -5000}`) against the unfixed code first, then confirms the fix
- [x] Covers the zero-balance edge case explicitly (legitimate value, must not be rejected)
- [x] Regression test confirms the default positive balance is unaffected
- [x] `pytest tests/unit/mcp/test_finstripe.py` — 3/3 passing

### CODE REVIEW

**Comment by Deez-Automations** [comment]
Addressed the Copilot review feedback:

- **Type safety**: real catch — `mock_balance < 0` would raise an unhandled `TypeError` if `mock_balance` were `None` or non-numeric. `server_config` is user-controllable JSON, so this was genuinely reachable, not theoretical. Added a type guard (excluding `bool`, since it's a subclass of `int` in Python) before the comparison, returning a clear error for invalid types too. Added tests for both `None` and a non-numeric string.
- **Docstring tense**: fixed — updated to past tense now that the check exists.

5/5 tests passing after both changes.

File: finbot/mcp/servers/finstripe/server.py [hunk 1]
```diff
@@ -120,6 +120,8 @@ def get_account_balance(account_id: str) -> dict[str, Any]:
         Returns the current available and pending balance for the specified account.
         """
         mock_balance = config.get("mock_balance", DEFAULT_CONFIG["mock_balance"])
+        if mock_balance < 0:
+            return {"error": "mock_balance is invalid: balance cannot be negative"}
```
[the remark this answers is not in the document]
Comment (Deez-Automations) [reply finbot/mcp/servers/finstripe/server.py:126]: Already handled — there's an isinstance(mock_balance, bool) or not isinstance(..., (int, float)) guard right before the < 0 check, returns a clean error instead of raising.

File: tests/unit/mcp/test_finstripe.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+GitHub issue #329 (Bug_120_MUST_FIX, MCP-BAL-005): get_account_balance
+reads mock_balance straight from server_config with no validation --
+a negative value is returned as-is as the account's available_balance.
+Since agents (e.g. PaymentsAgent) reason over this value when deciding
+whether a payment is affordable, a poisoned config with a negative
+balance can confuse those decisions.
+
+Verified against source before writing anything: finbot/mcp/servers/
+finstripe/server.py's get_account_balance (create_finstripe_server) has
+no bounds check on mock_balance at all.
```
[the remark this answers is not in the document]
Comment (Deez-Automations) [reply tests/unit/mcp/test_finstripe.py]: Already past tense in the current docstring — 'had no bounds check... before this fix'. Good catch on the pattern though, would've been a real issue.
